gma/src/ios/ump/consent_info_internal_ios.mm

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "gma/src/ios/ump/consent_info_internal_ios.h"
+
+#include "app/src/thread.h"
+#include "app/src/util_ios.h"
+
+namespace firebase {
+namespace gma {
+namespace ump {
+namespace internal {
+
+ConsentInfoInternalIos* ConsentInfoInternalIos::s_instance = nullptr;
+firebase::Mutex ConsentInfoInternalIos::s_instance_mutex;
+
+
+// This explicitly implements the constructor for the outer class,
+// ConsentInfoInternal.
+ConsentInfoInternal* ConsentInfoInternal::CreateInstance() {
+  return new ConsentInfoInternalIos();
```

Comment:
It feels weird that this isn't returning s_instance.  Maybe add an assert that s_instance is null, cause I'm guessing calling this multiple times would cause issues.

Reply:
ConsentInfo is a singleton so it should never create more than one of these - I'll add some code to return s_instance if it's valid, rather than an assert.